src/main.cpp

```diff
@@ -690,15 +691,31 @@ void setup()
 #endif
 
 #ifdef ARCH_RASPBERRY_PI
-    PiHal *RadioLibHAL = new PiHal(1);
-    if (!rIf) {
-        rIf = new SX1262Interface((LockingArduinoHal *)RadioLibHAL, 21, 16, 18, 20);
-        if (!rIf->init()) {
-            LOG_WARN("Failed to find SX1262 radio\n");
-            delete rIf;
-            rIf = NULL;
-        } else {
-            LOG_INFO("SX1262 Radio init succeeded, using SX1262 radio\n");
+    if (settingsMap[use_sx1262]) {
+        if (!rIf) {
+            PiHal *RadioLibHAL = new PiHal(1);
```

Comment:
where is this memory released if the radio is not sx1262?

Reply:
Honestly, if we can't init the radio as requested by the config, we should exit().